File: Tree/590.N-aryTree Postorder Traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
    const result =  [];
    (function dfs(node) {
        if (node) {
            if (node.children) {
                node.children.forEach(i => dfs(i));
            }
            result.push(node.val);
        }
    })(root);
    return result;
};

var postorder = function(root) {
    const stack = [root];
    const result = [];
    while(stack.length) {
        const node = stack.pop();
        if (node) {
            result.unshift(node.val);
            node.children.forEach(i => stack.push(i));
        }
    }
    return result;
};

File: Golang/Tree/590.N-ary-tree-postorder-traversal.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func postorder(root *Node) []int {
	ret := []int{}
	var helper func(node *Node)
	helper = func(node *Node) {
		if node != nil {
			for _, c := range node.Children {
				helper(c)
			}
			ret = append(ret, node.Val)
		}
	}
	helper(root)
	return ret
}
